expr/pool.go

```diff
@@ -12,3 +16,13 @@ var pointSlicePool *sync.Pool
 func Pool(p *sync.Pool) {
 	pointSlicePool = p
 }
+
+// GetPooledSliceAtLeastSize returns a 0-len slice that can hold at least minSize Points
+func GetPooledSliceAtLeastSize(minSize int) []schema.Point {
+	out := pointSlicePool.Get().([]schema.Point)
+	if cap(out) < minSize {
+		// TODO - return `out` to pool? Could just fetch it again next iteration
+		out = make([]schema.Point, 0, minSize)
```

Comment:
so in this case, out is not fetched out of the pool, it is more like calling make() locally

Reply:
yes